### DISCUSSION TITLE
Authentication with Bearer token

### DISCUSSION THREAD

**Discussion by tostenzel** [encode/httpx#1849] [Q&A] [answered]
(httpx version 0.16.1)

Hi guys, great package!

When I try to replace `aiohttp` by `httpx` I encounter the following problem:

When using `aiohttp` and requesting data from some WebAPI, it is sufficient to provide a Bearer token in the headers for some post request with

`headers = { 'Authorization': "Bearer xyz"}`

However, for the same url, data and additional header key-value pairs, this does not work with `httpx`.

Does somebody know how this can work?

**Comment by lovelydinosaur** [comment]
> However, for the same url, data and additional header key-value pairs, this does not work with `httpx`.

Can you show what you're doing and what behaviour you're seeing? (Keep your example absolutely as minimal as possible)

There's nothing wrong with how you've shown the headers there. One possibility I can think of here would be if you're requesting a URL and getting a redirect to a *different origin* then the Authorization headers would be stripped for the subsequent request.

**Reply by lovelydinosaur** [reply]
Similar discussion... #1846

**Comment by tostenzel** [comment, marked as answer]
Hi @tomchristie ! Thanks for responding that quickly!

The problem is now solved: I am directed via a proxy server and did not set `os.environ["NO_PROXY"] = "*"`. It seems as if aiohttp sets this by default.
